src/modules/kvs-watch/kvs-watch.c

```diff
@@ -1249,6 +1228,16 @@ static void namespace_getroot_continuation (flux_future_t *f, void *arg)
     watcher_respond_ns (nsm);
 }
 
+/* event.subscribe response for initial namespace creation */
+static void namespace_event_subscribe_continuation (flux_future_t *f, void *arg)
+{
+    struct ns_monitor *nsm = arg;
+
+    flux_future_destroy (f);
+    nsm->eventsubf = NULL;
+    nsm->subscribed = true;
+}
+
```

Comment:
This should probably be checking for error before destroying the future, and then on error, fail the watch request and destroy the nsm.

Alternatively, just send the subscribe request with the FLUX_RPC_NORESPONSE flag and destroy the future right there with a comment that notes that the unlikely failure is not handled.